x509/codesigning_cert_validations.go

```diff
@@ -0,0 +1,179 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package x509
+
+import (
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/internal/oid"
+)
+
+// ValidateCodeSigningCertChain takes an ordered code signing certificate chain
+// and validates issuance from leaf to root
+// Validates certificates according to this spec:
+// https://github.com/notaryproject/notaryproject/blob/main/specs/signature-specification.md#certificate-requirements
+func ValidateCodeSigningCertChain(certChain []*x509.Certificate, signingTime *time.Time) error {
+	if len(certChain) < 1 {
+		return errors.New("certificate chain must contain at least one certificate")
+	}
+	expectedLeafEku := x509.ExtKeyUsageAny
```

Comment:
Do you still need this since you have `validateCodeSigningLeafCertificate`?

Reply:
Removed.